[receiver/elasticsearch]: add metrics for merge operations with total shard aggregation

aboguszewski-sumo opened the pull request:
**Description:** 
Three new metrics have been added. One was already present in `metadata.yaml`, but was not emitted.

**Link to tracking Issue:** #14635

**Testing:**
Unit and integration

**Documentation:** 
Generated with `mdatagen`.

aboguszewski-sumo committed:
feat: add metrics for merge operations with total shard aggregation

djaglowski commented:
Would you mind splitting off a PR for the one that was mistakenly omitted? This is a bug and should be accepted separately.

The two new metrics seems reasonable to me.

aboguszewski-sumo commented:
I added the tests.

djaglowski commented on receiver/elasticsearchreceiver/metadata.yaml:
```diff
@@ -758,3 +758,21 @@ metrics:
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
```
Let's make these new metrics disabled by default.

aboguszewski-sumo commented:
@djaglowski I changed the metrics to disabled by default. There is one thing I've changed in the tests due to that. I decided that it would be the best if these metrics were enabled only in one test (I picked `full` because it seemed the most suitable one). There are two reasons for doing so:
1. This way we test the correctness of the metric, but also make sure that it is disabled by default.
2. Rebases will be much easier to be done. To be honest, the most annoying thing in working on these metrics is modifying the files with test results (and in case of this receiver, there are 5 files in total). If only one test file has to be modified with each PR, the rebases can go more smoothly.
I'd like to follow this convention for other metrics that are going to be disabled by default (I'll be fixing the PRs gradually), so if you have any objection, please raise it now.

djaglowski commented:
@aboguszewski-sumo, I agree with your reasoning on the test files. This should be a good pattern to follow going forward.
